src/main/java/com/_up/megastore/services/implementations/SizeService.java

```diff
@@ -44,6 +46,18 @@ public SizeResponse updateSize(UUID sizeId,UpdateSizeRequest updateSizeRequest){
         return SizeMapper.toSizeResponse(sizeRepository.save(size));
     }
 
+    @Override
+    public SizeResponse readSize(UUID sizeId){
+        Size size = findSizeByIdOrThrowException(sizeId);
+        return SizeMapper.toSizeResponse(size);
+    }
+
+    @Override
+    public Page<SizeResponse> readAllSizes(int page, int pageSize, String name){
+        Pageable pageable = PageRequest.of(page, pageSize);
+        return sizeRepository.findSizeByDeletedIsFalseAndNameContainingIgnoreCase(name,pageable).map(SizeMapper::toSizeResponse);
```

Comment:
Para dejar más limpio el código podrías hacer lo siguiente:

```suggestion
        return sizeRepository.findSizeByDeletedIsFalseAndNameContainingIgnoreCase(name,pageable)
              .map(SizeMapper::toSizeResponse);
```
Y no tenemos todo en la misma línea. De todos modos no te voy a pedir que lo cambies, es una sugerencia